Chilli not builds for new Kernel

```
OpenWrt-libtool: compile:  mips-openwrt-linux-musl-gcc -DHAVE_CONFIG_H -I. -I.. -I../json -I/home/eri/openwrt/staging_dir/target-mips_34kc_musl-1.1.15/usr/include -I/home/eri/openwrt/staging_dir/target-mips_34kc_musl-1.1.15/include -I/home/eri/openwrt/staging_dir/toolchain-mips_34kc_gcc-5.3.0_musl-1.1.15/usr/include -I/home/eri/openwrt/staging_dir/toolchain-mips_34kc_gcc-5.3.0_musl-1.1.15/include/fortify -I/home/eri/openwrt/staging_dir/toolchain-mips_34kc_gcc-5.3.0_musl-1.1.15/include -D_GNU_SOURCE -Wall -Werror -fno-builtin -fno-strict-aliasing -fomit-frame-pointer -funroll-loops -pipe -I../bstring -DDEFCHILLICONF=\"/etc/chilli.conf\" -DDEFPIDFILE=\"/var/run/chilli.pid\" -DDEFSTATEDIR=\"/var/run\" -DSBINDIR=\"/usr/sbin\" -Os -pipe -mno-branch-likely -mips32r2 -mtune=34kc -fno-caller-saves -fno-plt -fhonour-copts -Wno-error=unused-but-set-variable -Wno-error=unused-result -msoft-float -mips16 -minterlink-mips16 -iremap /home/eri/openwrt/build_dir/target-mips_34kc_musl-1.1.15/coova-chilli-2016:coova-chilli-2016 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro -fpic -MT chilli.lo -MD -MP -MF .deps/chilli.Tpo -c chilli.c  -fPIC -DPIC -o .libs/chilli.o
In file included from /home/eri/openwrt/staging_dir/toolchain-mips_34kc_gcc-5.3.0_musl-1.1.15/include/net/ethernet.h:10:0,
                 from system.h:173,
                 from chilli.h:24,
                 from chilli.c:21:
/home/eri/openwrt/staging_dir/toolchain-mips_34kc_gcc-5.3.0_musl-1.1.15/include/netinet/if_ether.h:97:8: error: redefinition of 'struct ethhdr'
 struct ethhdr {
        ^
In file included from system.h:124:0,
                 from chilli.h:24,
                 from chilli.c:21:
/home/eri/openwrt/staging_dir/toolchain-mips_34kc_gcc-5.3.0_musl-1.1.15/include/linux/if_ether.h:140:8: note: originally defined here
 struct ethhdr {
        ^
Makefile:936: recipe for target 'chilli.lo' failed

```

--- a/src/system.h
+++ b/src/system.h
@@ -169,7 +169,7 @@
 #include <net/if_tun.h>
 #endif
 
-#ifdef HAVE_NET_ETHERNET_H
+#if defined(HAVE_NET_ETHERNET_H) && !defined(__linux__)
 #include <net/ethernet.h>
 #endif
 

